kiali-server/values.yaml

```diff
@@ -87,7 +87,7 @@ deployment:
   tolerations: []
   version_label: ${HELM_IMAGE_TAG} # v1.39 # v1.39.0 # see: https://quay.io/repository/kiali/kiali?tab=tags
   view_only_mode: false
-
+  csiVolumes: []
```

Comment:
Example yaml with a defintion 
```yaml
deployment:
  csiVolumes:
    csi:
      driver: secrets-store.csi.k8s.io
      readOnly: true
      volumeAttributes:
        secretProviderClass: kiali-secretprovider
```

Reply:
keep everything consistent with the way the rest of the current config is formatted - use snake_case for setting names.

here's our developer checklist for more things to keep in mind: https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#how-to-change-resources-or-configuration

Reply:
> keep everything consistent with the way the rest of the current config is formatted - use snake_case for setting names.

I say that, however, if I'm not mistaken, is everything underneat csiVolumes.csi all the yaml that gets copied directly in the volume definition of the Deployment? If so, they can probably stay since the "csi" is just defining the direct Deployment volume definition.

And I also think it looks like csiVolumes.csi is redundant. Why not just:

```
deployment.
  csi_volume:
    driver: ...
    readOnly: ...
    volumeAttributes: ...
```

And it is one volume, right? I don't see "csi" in your code as an array.. so `csi_volume` seems appropriate (as opposed to the plural `csi_volumes`.)

Reply:
Oh, I think because you might need an "enabled" flag in here. Perhaps this is what we want?

```
csi:
  enabled: true | false
  csi_volume:
  ... your volume YAML here...
```

Reply:
note: you don't HAVE to have an enabled flag - you could just say "if deployment.csi_volume" is not defined, assume there is no CSI and don't install any volume". That way you can just leave it with `deployment.csi_volume` - that is also backward compatible since no one has deployment.csi_volume  in their config today so it will jsut work as if it isn't defined/enabled.

Reply:
Thanks! Everything under the `csi` stanza maps to the csi [here](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume/#Volume). That should future-proof the helm chart against changes to the csi spec. If you prefer I template out every option, I could do that as well.

Reply:
> And it is one volume, right? I don't see "csi" in your code as an array.. so `csi_volume` seems appropriate (as opposed to the plural `csi_volumes`.)

Correct it will most likely be a single volume. I made it an array to allow for additional mounts if required but maybe that is a premature optimization.